spec/models/service_template_ansible_playbook_spec.rb

```diff
@@ -202,7 +202,23 @@
     end
   end
 
-  describe '#update_catalog_item' do
+  context '#update_for_new_template' do
+    it 'returns immediately if we do not have a playbook_id and job_template is not nil' do
+      service_template = prebuild_service_template
+      info = catalog_item_options_two[:config_info][:retirement]
+      update_resp = service_template.send(:update_for_new_template, info, :retirement, 'test_name', 'test', 'system')
+      expect(update_resp).to be_nil
+    end
+
+    it 'adds a new configuration_template if a playbook_id is passed in and the job_template is nil' do
+      service_template = prebuild_service_template(:job_template => false)
+      info = catalog_item_options_two[:config_info][:retirement]
+      allow_any_instance_of(Hash).to receive(:fetch_path).and_return(job_template)
```

Comment:
I'm open to suggestions on this as I realize `allow_any_instance_of` is generally a no no.